File: PR1/PR1_Ej3/DisplayPedidos.py
from ObservadorPedido import ObservadorPedido
import os
class DisplayPedidos(ObservadorPedido):
    def __init__(self):
        self.historial = []
        if os.path.exists("historial.txt"):
            os.remove("historial.txt")
        self.f = open("historial.txt", "x")
        self.f.close()

    def update(self, pedido):
        self.historial.append(pedido)
        print(f"\nEl pedido {pedido.idPedido} está listo!")
        self.saveToTxT(pedido)

    def saveToTxT(self, pedido):
        self.f = open("historial.txt", "a")
        self.f.write(pedido.__str__())
        self.f.close()

File: PR2/pantalla_pedidos_hamburgueseria/lib/model/DisplayPedidos.dart
import 'ObservadorPedido.dart';
import 'Pedido.dart';
import 'package:flutter/material.dart';

import 'dart:convert';
import 'package:http/http.dart' as http;

class DisplayPedidos implements ObservadorPedido {
  List<Pedido> historial = [];
  final String apiUrl = "http://localhost:3000/pedidos";

  late ScaffoldMessengerState _scaffoldMessengerState; // Referencia al ScaffoldMessenger
  late Function() _actualizarHistorial; // Agregar una función de actualización

  DisplayPedidos(){}

  DisplayPedidos.Parametros(this.historial, this._actualizarHistorial);

  void init(BuildContext context) {
    _scaffoldMessengerState = ScaffoldMessenger.of(context);
  }

  void mostrarSnackBar(BuildContext context, String mensaje) {
    final snackBar = SnackBar(
      content: Text(mensaje),
      duration: Duration(seconds: 5), // Duración del snackbar
      backgroundColor: Colors.redAccent,
    );
    _scaffoldMessengerState.showSnackBar(snackBar);
  }

  @override
  void update(Pedido pedido, BuildContext? context) {
    if (context != null){
      if (!_scaffoldMessengerState.mounted) {
        // Comprobar si el ScaffoldMessenger sigue montado antes de mostrar el SnackBar
        return;
      }
      mostrarSnackBar(context, 'El pedido ${pedido.idPedido}\n $pedido está listo');
      _actualizarHistorial();
    }
    historial.add(pedido);
  }

  //Parte API Rest
  Future<void> cargarPedidos(String usuarios) async {
    final response = http.get(Uri.parse('$apiUrl?usuario=$usuario'));
    if (response.statusCode == 200) {
      List<dynamic> pedidosJson = json.decode(respose.body);

      historial.clear();
      historial.addAll(pedidosJson.map((json) => Pedido.fromJson(json)).toList());
    } else {
      throw Exception('Fallo al cargar pedidos');
    }
  }


}